artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/cursor/impl/PageCursorProviderImpl.java

```diff
@@ -599,6 +600,29 @@ private long checkMinPage(Collection<PageSubscription> cursorList) {
 
    }
 
+   private void deliverIfNecessary(Collection<PageSubscription> cursorList) {
+      long minPage = Long.MAX_VALUE;
+      PageSubscription slowSubscription = null;
+      int nonEmptyCursorNum = 0;
+
+      for (PageSubscription cursor : cursorList) {
+         long firstPage = cursor.getFirstPage();
+
+         // the cursor will return -1 if the cursor is empty
+         if (firstPage >= 0) {
+            nonEmptyCursorNum++;
+            if (firstPage < minPage) {
+               minPage = firstPage;
+               slowSubscription = cursor.getQueue().getMessageCount() == 0 ? cursor : null;
+            }
+         }
+      }
+
+      if (nonEmptyCursorNum > 1 && slowSubscription != null) {
+         slowSubscription.getQueue().deliverAsync();
```

Comment:
Just to understand the fix: given that QueueImpl::deliverAsync will trigger QueueImpl::checkDepage that will schedule an async task QueueImpl::DepageRunner how do you know that it will be finished time in order to have the PageSubscription completed?

Reply:
@wy96f man, you're good!

Reply:
@wy96f  how do I get to meet you? :) IRC / talk... etc? you have been providing great fixes.

Reply:
@wy96f  @clebertsuconic totally agree!

Reply:
@clebertsuconic Thank you man. We're heavily using artemis in our company. It's a great product and thanks for your work. We are a team: shoukunhuai, me, shoukunhuai. We will keep on developing the code :)